types/field_type.go

```diff
@@ -39,7 +39,8 @@ type FieldType struct {
 	Charset string
 	Collate string
 	// Elems is the element list for enum and set type.
-	Elems []string
+	Elems                  []string
+	NegativeUnsignedBeZero bool
```

Comment:
yes~ because in union context cast(negative unsigned be zero) is differ to cast in normal(negative unsigned be int.Max)...so we need a flag.

we can add it in field_type or cast expression/executor, and it seems previous way's modification is less.

yes, it is not very comfortable for mee too, or have any other good way ~ ^ ^ ?